src/Compilers/Core/Portable/VersionHelper.cs

```diff
@@ -65,18 +66,43 @@ private static bool TryParse(string s, bool allowWildcard, ushort maxValue, out
 
             if ((hasWildcard && elements.Length < 3) || elements.Length > 4)
             {
-                version = null;
+                version = new Version(0, 0, 0, 0);
                 return false;
             }
 
             ushort[] values = new ushort[4];
             int lastExplicitValue = hasWildcard ? elements.Length - 1 : elements.Length;
+            bool parseError = false;
             for (int i = 0; i < lastExplicitValue; i++)
             {
-                if (!ushort.TryParse(elements[i], NumberStyles.None, CultureInfo.InvariantCulture, out values[i]) || values[i] > maxValue)
+                
+                if (allowPartialParse)
                 {
-                    version = null;
-                    return false;
+                    //Shorten the version string until we are at the shortest possible string that can be represented as a ushort
+                    while (!ushort.TryParse(elements[i], NumberStyles.None, CultureInfo.InvariantCulture, out values[i]))
```

Comment:
It looks like on the `allowPartialParse` path we are no longer doing the `values[i] > maxValue` check.  I believe that check should still be there.